### PR TITLE
Added new way of component creation and reg 

### PR DESCRIPTION
**Pull request by Revolyssup**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,3 +5,7 @@ updates:
   schedule:
     interval: daily
   open-pull-requests-limit: 10
+  reviewers:
+    - Revolyssup
+    - DelusionalOptimist
+    - utkarsh-pro
```
Comment (leecalcote) [inline .github/dependabot.yml:11]: Did each of these individuals confirm their assignment?

Comment (Revolyssup) [reply .github/dependabot.yml:11]: Yes

File: .github/workflows/update.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      run: |
+          wd=$(pwd)
+          mkdir -p ~/.meshery/bin
+          cd ~/.meshery/bin
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+          export NVM_DIR="$HOME/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+          nvm install v14.15.3
+          nvm install-latest-npm
+          git clone https://github.com/layer5io/kubeopenapi-jsonschema util
```
Comment (leecalcote) [inline .github/workflows/update.yaml:28]: Is this temporary until switching over to running an instance of the adapter itself?

File: build/config.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/layer5io/meshkit/utils/manifests"
+	smp "github.com/layer5io/service-mesh-performance/spec"
+)
+
+var DefaultGenerationMethod string
+var DefaultGenerationURL string
+var LatestVersion string
+var WorkloadPath string
+
+//Should stay here
```
Comment (leecalcote) [inline build/config.go:19]: What should?

File: go.mod [hunk 4]
```diff
@@ -12,8 +12,8 @@ replace (
 
 require (
 	github.com/aspenmesh/istio-vet v0.0.0-20200806222806-9c8e9a962b9f
-	github.com/layer5io/meshery-adapter-library v0.5.0
-	github.com/layer5io/meshkit v0.5.0
+	github.com/layer5io/meshery-adapter-library v0.5.2
+	github.com/layer5io/meshkit v0.5.1
```
Comment (leecalcote) [inline go.mod:16]: Nice

File: build/config.go (hunk 3 above)
Comment (Revolyssup) [reply build/config.go:19]: Changed it

File: .github/workflows/update.yaml (hunk 2 above)
Comment (Revolyssup) [reply .github/workflows/update.yaml:28]: It still would need kubeopenapi-jsonschema. We are using this to build the binary

File: main.go [hunk 5]
```diff
@@ -170,62 +169,55 @@ func registerDynamicCapabilities(port string, log logger.Handler) {
 		<-ticker.C
 		registerWorkloads(port, log)
 	}
-
 }
 func registerWorkloads(port string, log logger.Handler) {
-	var url string
-	var gm string
+	log.Info("Registering latest components dynamically with Meshery server")
```
Comment (leecalcote) [inline main.go:174]:
```suggestion
	log.Info("Registering latest components with Meshery Server")
```

File: main.go [hunk 6]
```diff
@@ -170,62 +169,55 @@ func registerDynamicCapabilities(port string, log logger.Handler) {
...
+		DirName: version,
+		Config:  build.NewConfig(version),
+	})
+
+	if err != nil {
+		log.Info("Failed to generate components for version "+version, "ERR: ", err.Error())
+		return
+	}
+	//The below log is checked in the workflows. If you change this log, reflect that change in the workflow where components are generated
+	log.Info("Component Creation completed for version ", version)
```
Comment (leecalcote) [inline main.go:207]:
```suggestion
	log.Info("Component creation completed for version ", version)
```

File: main.go [hunk 7]
```diff
@@ -170,62 +169,55 @@ func registerDynamicCapabilities(port string, log logger.Handler) {
...
+
+	//Now we will register in case
+	log.Info("Registering workloads with Meshery Server for version...", version)
```
Comment (leecalcote) [inline main.go:210]:
```suggestion
	log.Info("Registering workloads with Meshery Server for version ", version)
```

File: .github/dependabot.yml (hunk 1 above)
Comment (leecalcote) [reply .github/dependabot.yml:11]: Wonderful. Thank you for getting us organized. ✔️

File: .github/workflows/update.yaml (hunk 2 above)
Comment (leecalcote) [reply .github/workflows/update.yaml:28]: Makes much sense that the binary will still be around. What I'm asking is why we're still scripting around the binary and not calling the binary from what will be a Golang-wrapped approach consistently used across every Adapter and Server.

**Comment by leecalcote** [comment]
Waiving DCO.

File: .github/workflows/update.yaml (hunk 2 above)
Comment (leecalcote) [reply .github/workflows/update.yaml]: To be clear, components will still be generated in a workflow and persisted in this repo. Is this work completed already?

Comment (leecalcote) [reply .github/workflows/update.yaml]: It seems that the new workflow covers this now, is that right?

**Comment by Revolyssup** [comment]
@leecalcote  I wasnt able to reply below. Yes the components will be generated and persisted.

**Comment by leecalcote** [comment]
> @leecalcote I wasnt able to reply above. Yes the components will be generated and persisted.

Super. ☑️

**Review by leecalcote — APPROVED** [review]

**Pull request merged**